docs/feature_flags.md

```diff
@@ -0,0 +1,57 @@
+# Feature Flags
```

Comment:
WDYT about adding a section to explain how FFs enable Continuous Deployment to our dev teams? This is a good reference to add: https://about.gitlab.com/handbook/product-development-flow/feature-flag-lifecycle/#the-benefits-of-feature-flags